pkg/sbom/nodelist.go

```diff
@@ -749,3 +817,78 @@ func (nl *NodeList) NodeSiblings(id string) *NodeList {
 
 	return nodelist
 }
+
+// NodeDescendants traverses the NodeList graph starting at the node specified
+// by id and returns a new node list with elements related at a maximal distance
+// of maxDepth levels. If the specified id is not found, the NodeList will be
+// empty. Traversing the graph will stop if any of the related nodes is a RootNode.
+func (nl *NodeList) NodeDescendants(id string, maxDepth int) *NodeList {
+	rootIdx := nl.indexRootElements()
+	edgeIdx := nl.indexEdges()
+	startNode := nl.GetNodeByID(id)
+	if startNode == nil {
+		return &NodeList{}
+	}
+
+	nl2 := NodeList{
+		Nodes:        []*Node{},
+		Edges:        nl.Edges,
+		RootElements: []string{startNode.Id},
+	}
+
+	siblings := nodeIndex{}
+
+	var loopNodes []*Node
+	newLoopNodes := []*Node{}
+
+	for i := 0; i < maxDepth; i++ {
+		if i == 0 {
+			fmt.Fprintf(os.Stdout, "Nuevo\n")
```

Comment:
I get happy little 

```
Nuevo
Nuevo
Nuevo
Nuevo
Nuevo
```

output from go test when I run it now.  Probably don't need this :)

Reply:
Ah sorry I left this debugger string. Thanks, fixed!